### PR TITLE
Show current scene file in FileSystem dock

### LINKED ISSUE #19870: Add 'scroll to the active scene' feature to the file system view
**Godot version:**
`master` / 0ffec7daf

**OS/device including version:**
_Manjaro Linux 17.1_

**Issue description:** _(This issue is a feature request)_
In many applications whose main window consists of a navigator tree and a content view (i.e. IDEs), there is a button on top of the navigator called something like _'Scroll from source'_ or _'Show in hierarchy'_ with which you can quickly reveal the location of the item you are currently viewing in the content view within the hierarchy.

Currently, Godot can optionally remembers scenes that were opened in the last session and restore them when reopened, but the file system hierarchy always shows as collapsed initially. So, it may take some effort to navigate to the directory which contains the scene loaded in each of the restored tabs when they are located deep inside the hierarchy.

If we can add such a button on top of the file system view as mentioned above, we can quickly navigate to the location where the current scene can be found without having to manually expand the tree nodes multiple times, thus improving the user experience.

### PR DESCRIPTION
**Pull request by volzhs**
close #19870

show_scene_file

### CODE REVIEW

**Comment by mysticfall** [comment]
Oh my... that was fast! Thanks much! :)

File: editor/filesystem_dock.cpp [hunk 1]
```diff
@@ -1864,6 +1876,13 @@ FileSystemDock::FileSystemDock(EditorNode *p_editor) {
 	button_favorite->set_focus_mode(FOCUS_NONE);
 	toolbar_hbc->add_child(button_favorite);
 
+	button_show = memnew(Button);
+	button_show->set_flat(true);
+	button_show->connect("pressed", this, "_show_current_scene_file");
+	toolbar_hbc->add_child(button_show);
+	button_show->set_focus_mode(FOCUS_NONE);
+	button_show->set_tooltip("Show current scene file.");
```
Comment (guilhermefelipecgs) [inline editor/filesystem_dock.cpp]: TTR() missing here.

Comment (volzhs) [reply editor/filesystem_dock.cpp]: @guilhermefelipecgs oops.

Comment (guilhermefelipecgs) [reply editor/filesystem_dock.cpp]: :+1:

**Comment by Zylann** [comment]
While this is handy, I wonder why this isn't just made with a context menu?
The eye icon restricts the functionality to scenes, while a context menu can do that by clicking on scene tabs, resources, or even script tabs.

**Comment by volzhs** [comment]
@Zylann I looked for context menu for scene tab first if it has, but found nothing for it.
so, I just made like this as #19870 request.
PR for consistency and usability for overall is welcome.

**Pull request merged**